src/Nethermind/Nethermind.Crypto/BlsSigner.cs

```diff
@@ -23,96 +23,132 @@ public static class BlsSigner
     private const int InputLength = 64;
 
     [SkipLocalsInit]
+    // buf must be of size G2.Sz
+    public static Signature Sign(Span<long> buf, Bls.SecretKey sk, ReadOnlySpan<byte> message)
+    {
+        if (buf.Length != G2.Sz)
+        {
+            throw new ArgumentException($"Signature buffer {nameof(buf)} must be of size {G2.Sz}.");
+        }
+
+        G2 p = new(buf);
+        Signature s = new(p);
+        s.Sign(sk, message);
+        return s;
+    }
+
     public static Signature Sign(Bls.SecretKey sk, ReadOnlySpan<byte> message)
+        => Sign(new long[G2.Sz], sk, message);
```

Comment:
Can we stackalloc here?

Reply:
I made two versions of `Sign`, one takes a buffer as input so you can stackalloc, this one uses new

Reply:
but why can't it: `Sign(stackalloc long[G2.Sz], sk, message);` ?